src/server/server.js

```diff
@@ -77,21 +223,30 @@ function streamAppHostHtml(request, response) {
             }
         }).pipe(response);
     }).done();
-}
+};
 
-function streamSimHostHtml(request, response) {
+/**
+ * @param {object} request
+ * @param {object} response
+ * @private
+ */
+SimulationServer.prototype._streamSimHostHtml = function (request, response) {
     // If we haven't ever prepared, do so before we try to generate sim-host, so we know our list of plugins is up-to-date.
     // Then create sim-host.js (if it is out-of-date) so it is ready when it is requested.
-    prepare.prepare().then(function () {
-        return simFiles.createSimHostJsFile();
-    }).then(function () {
+    var project = this._project;
+
+    project.prepare().then(function () {
```

Comment:
I prefer the indentation style you propose, it is better for reading it. In many places it was intended like the code in this review. @dlebu @guillaumejenkins @TimBarham @MSLaguana what do you think if we go with this indentation style that @busykai is proposing for promises chain? In my opinion, it is better for readability and easily understanding/following the promises chain.